python/pyspark/sql/dataframe.py

```diff
@@ -470,7 +470,7 @@ def sampleBy(self, col, fractions, seed=None):
         +---+-----+
 
         """
-        if not isinstance(col, str):
+        if not isinstance(col, basestring):
```

Comment:
Is there some harm in allowing unicode here though? If my column is `'a'` and I call `sampleBy(u'a')` it will work after this change, otherwise it will throw an error. I think it's better to treat `'a'` and `u'a'` as equivalent...